### PR TITLE
HTS-481: Updated eligibility check stub to reflect all responses DES …

### PR DESCRIPTION
**Pull request by MarcusQA**
…can send

### CODE REVIEW

File: app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala [hunk 1]
```diff
@@ -42,32 +43,55 @@ class EligibilityCheckController extends BaseController {
...
+        Some(ineligibleResult(getReasonCodeFromNino(nino)))
+      } // Start NINO with AC to specify an existing account holder (with reason code 1)
+      else if (nino.startsWith("AC")) {
         Some(alreadyHasAccountResult)
-      } else if (nino.toUpperCase().startsWith("NA")) {
-        ineligibleReasonGen.seeded(nino)
-      } else {
-        eligibleReasonGen.seeded(nino)
+      } // Start NINO with EE to specify an invalid result code
+      else if (nino.startsWith("EE")) {
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala]: why do need this ?  this means that for specific ninos DES always returns invalid result codes ?

File: app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala [hunk 2]
```diff
@@ -16,19 +16,20 @@
...
+import scala.util.Try
 
 class EligibilityCheckController extends BaseController {
 
   val resultMappings: Map[Int, String] = Map(
     1 → "Eligible to HtS Account",
     2 → "Ineligible to HtS Account",
-    3 → "HtS account already exists"
+    3 → "HtS account already exists",
+    4 → "INVALID RESULT WHICH DES SHOULD NEVER SEND"
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala]: i am not comfortable adding an invalid code here

File: app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala [hunk 3]
```diff
@@ -42,32 +43,55 @@ class EligibilityCheckController extends BaseController {
...
-    Gen.choose(2, 5) // scalastyle:ignore magic.number
-      .map{ i ⇒
-        val reason = reasonMappings.getOrElse(i, sys.error(s"Could not find ineligibility reason for code $i"))
-        EligibilityCheckResult("Ineligible to HtS Account", 2, reason, i)
-      }
-
   val alreadyHasAccountResult: EligibilityCheckResult =
     EligibilityCheckResult("HtS account already exists", 3, "HtS account was previously created", 1)
 
+  val invalidResultCode: EligibilityCheckResult =
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala]: no unit tests for all these changes ?

File: app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala [hunk 4]
```diff
@@ -42,32 +43,55 @@ class EligibilityCheckController extends BaseController {
...
+  }
+
+  def ineligibleResult(reasonCode: Int): EligibilityCheckResult = {
+    val reason = reasonMappings.getOrElse(reasonCode, sys.error(s"Could not find eligibility reason for code $reasonCode"))
+    EligibilityCheckResult("Ineligible to HtS Account", 2, reason, reasonCode)
+  }
+
+  def getReasonCodeFromNino(nino: String): Int =
+    Try(nino.substring(3, 4).toInt)
+      .getOrElse(sys.error(s"Error getting reason code from fourth character of NINO $nino"))
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala]: does `nino.substring(3, 4)` this gives `fourth character of NINO ` really ?  or it gives fourth and fifth characters too ?

File: app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala [hunk 5]
```diff
@@ -42,32 +43,55 @@ class EligibilityCheckController extends BaseController {
...
+      if (nino.toUpperCase().startsWith("EL")) {
+        Some(eligibleResult(getReasonCodeFromNino(nino)))
+      } // Private BETA:
+      // Start NINO with NE02 to specify an eligible applicant in receipt of WTC (with reason code 2)
+      // Start NINO with NE03 to specify an eligible customer in receipt of WTC (with reason code 3)
+      //
+      // After private BETA:
+      // Start NINO with NE06 to specify an eligible applicant in receipt of UC (with reason code 6)
+      // Start NINO with NE08 to specify an eligible applicant in receipt of WTC and UC (with reason code 8)
+      else if (nino.toUpperCase().startsWith("NE")) {
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala]: the above doc says for NE numbers signifies  `to specify an eligible applicant ` but down below you are returning `ineligibleResult ` ?

File: app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala (hunk 2 above)
Comment (MarcusQA) [reply app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala]: I'd like the ability to check that we handle incorrect values from DES gracefully.

File: app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala (hunk 3 above)
Comment (MarcusQA) [reply app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala]: Can I seek your help with some unit tests, please?

File: app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala (hunk 4 above)
Comment (MarcusQA) [reply app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala]: Just the fourth character.

File: app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala (hunk 5 above)
Comment (MarcusQA) [reply app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala]: Good spot - I've corrected this.

File: app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala (hunk 1 above)
Comment (MarcusQA) [reply app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala]: Yes - I'd like the ability to check that we handle incorrect values from DES gracefully.

File: app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala (hunk 2 above)
Comment (sureshhmrc) [reply app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala]: the incorrect values are handled in the frontend accordingly

File: app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala (hunk 3 above)
Comment (sureshhmrc) [reply app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala]: sure, i will help you on monday ?

Comment (MarcusQA) [reply app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala]: Sounds good - thanks Suresh.

**Pull request merged**